sovtokenfees/sovtokenfees/main.py

```diff
@@ -1,93 +1,168 @@
+    return node
 
-    token_authnr = node.clientAuthNr.get_authnr_by_type(TokenAuthNr)
-    if not token_authnr:
-        raise ImportError('sovtoken plugin should be loaded, '  # noqa
-                          'authenticator not found')
-    token_req_handler = node.get_req_handler(ledger_id=TOKEN_LEDGER_ID)
-    if not token_req_handler:
-        raise ImportError('sovtoken plugin should be loaded, request '  # noqa
+
+def register_req_handlers(node, fees_tracker):
+    node.write_manager.register_req_handler(SetFeesHandler(node.db_manager,
+                                                           node.write_req_validator))
+
+    if XFER_PUBLIC not in node.write_manager.request_handlers:
+        raise ImportError('sovtoken plugin should be loaded, request '
                           'handler not found')
+    node.write_manager.remove_req_handler(XFER_PUBLIC)
+    node.write_manager.register_req_handler(XferFeeHandler(node.db_manager,
+                                                           node.write_req_validator))
+
+    for typ in list(node.write_manager.ledger_id_to_types[DOMAIN_LEDGER_ID]):
+        # Ugly hack, replace it with expanding register_req_handler method
+        # TODO: Additional functionality to request_manager ^^^
+        domain_fee_r_h = DomainFeeHandler(node.db_manager, fees_tracker)
+        domain_fee_r_h.txn_type = typ
+        node.write_manager.register_req_handler(domain_fee_r_h)
+
+    node.write_manager.register_req_handler(FeeTxnCatchupHandler(node.db_manager))
+    node.read_manager.register_req_handler(GetFeeHandler(node.db_manager))
+    gfs_handler = GetFeesHandler(node.db_manager)
+    node.read_manager.register_req_handler(gfs_handler)
+
+    node.write_manager.register_req_handler(AuthRuleFeeHandler(node.db_manager, gfs_handler))
+    node.write_manager.register_req_handler(AuthRulesFeeHandler(node.db_manager, gfs_handler))
+
+
+def register_batch_handlers(node, fees_tracker):
+    handlers = node.write_manager.batch_handlers[DOMAIN_LEDGER_ID]
+    node.write_manager.remove_batch_handler(DOMAIN_LEDGER_ID)
+
+    # Temp checks, remove after integration
+    if len(handlers) != 4 or not isinstance(handlers[-1], TsStoreBatchHandler) \
+            or not (handlers[-2], AuditBatchHandler):
+        raise LogicError
+
+    handlers.insert(handlers.index(handlers[-2]),
+                    DomainFeeBatchHandler(node.db_manager, fees_tracker))
+
+    for h in handlers:
+        if isinstance(h, (AuditBatchHandler, TsStoreBatchHandler)):
+            node.write_manager.register_batch_handler(h, ledger_id=DOMAIN_LEDGER_ID)
+        node.write_manager.register_batch_handler(h)
+    # TODO: Additional functionality to request_manager ^^^
+
+
+def set_callbacks(node):
+    set_post_catchup_callback(node)
+    set_post_added_txn_callback(node)
+
+
+def set_post_catchup_callback(node):
+    def postCatchupCompleteClbk(node):
+        token_tracker = node.db_manager.get_tracker(TOKEN_LEDGER_ID)
```

Comment:
For now, token_tracker is registered in db_manager. In future we can simplify this callback

Reply:
ok